common/persistence/visibility/visibility_manager_dual.go

```diff
@@ -27,15 +27,17 @@ package visibility
 import (
 	"context"
 
+	"go.temporal.io/server/common/dynamicconfig"
 	"go.temporal.io/server/common/namespace"
 	"go.temporal.io/server/common/persistence/visibility/manager"
 )
 
 type (
 	VisibilityManagerDual struct {
-		visibilityManager          manager.VisibilityManager
-		secondaryVisibilityManager manager.VisibilityManager
-		managerSelector            managerSelector
+		visibilityManager              manager.VisibilityManager
+		secondaryVisibilityManager     manager.VisibilityManager
+		managerSelector                managerSelector
+		visibilityEnableShadowReadMode dynamicconfig.BoolPropertyFn
```

Comment:
Here you don't need to prefix with `visibility`
```suggestion
		enableShadowReadMode dynamicconfig.BoolPropertyFn
```